tests/src/python/test_qgslayoutlegend.py

```diff
@@ -463,7 +458,7 @@ def testSymbolExpressionRender(self):
         group = legend.model().rootGroup().addGroup("Group [% 1 + 5 %] [% @layout_name %]")
         layer_tree_layer = group.addLayer(point_layer)
         counterTask = point_layer.countSymbolFeatures()
-        counterTask.waitForFinished() # does this even work?
+        counterTask.waitForFinished()
```

Comment:
Thanks for solving this, thought this might be an out of place fix for this PR.

Reply:
Yes, I was thinking about making a PR for the waitForFinished fix and maybe backporting it to LTR because it may produce some crashes.

Reply:
I would assume that core devs will suggest having that change as a standalone PR.

Reply:
Done https://github.com/qgis/QGIS/pull/32453